src/main/app/jurisdiction/connections.ts

```diff
@@ -56,12 +56,21 @@ const onlyApplicant1Domiciled = data => {
 export const allowedToAnswerResidualJurisdiction = (data: Partial<CaseWithId>): boolean => {
   return (
     (data.sameSex === Checkbox.Checked || data.divorceOrDissolution === DivorceOrDissolution.DISSOLUTION) &&
-    data.partnersLifeBasedInEnglandAndWales === YesOrNo.NO &&
-    !isDomiciled('your', data) &&
-    !isDomiciled('partners', data)
+    data.lastHabituallyResident === YesOrNo.NO &&
+    !previousConnectionMadeUptoLastHabituallyResident(data)
   );
 };
 
+export const previousConnectionMadeUptoLastHabituallyResident = (data: Partial<CaseWithId>): boolean => {
+  if (data.connections?.includes(JurisdictionConnections.APP_1_APP_2_LAST_RESIDENT) && data.connections?.length > 1) {
+    return true;
+  } else {
+    return !!(
+      !data.connections?.includes(JurisdictionConnections.APP_1_APP_2_LAST_RESIDENT) && data.connections?.length
+    );
+  }
```

Comment:
Might be worth adding a unit test to `src/main/app/jurisdiction/connections.test.ts` to cover these new scenarios